andela/ah-scorpion
# #159459856 Refactor Code To Meet README Instructions

**What does this PR do?**
It refactor's the code to meet the README instructions.

**Description of tasks to be completed**
- Version the API. From now on, add `v1` to the endpoints like so:
  - `GET api/v1/profiles/<username>`
  - `GET api/v1/user/`
  - `GET api/v1/articles/`
- Replace the registration endpoint with the more explicit `POST api/v1/users/signup`
- Reduce the use of Django's `url` function in favour of the `path` function to avoid writing regular expressions.
- Modify the authentication views such that the request does not need the top level keyword so as to be processed.
- Modify tests to meet the modification on the authentication views

**How can this be manually tested?**
 - Clone the repository from [here](https://github.com/andela/ah-scorpion/tree/ch-Refactor-Code-%23159459856) and check out the branch **ch-Refactor-Code-#159459856**
 - Activate your virtual environment and use the .env-sample to set the environment variables.
 - Use postman to test these endpoints.
    - `POST api/v1/users/signup`
    - `POST api/v1/users/login`
    - `GET api/v1/user/`
    - `PUT api/v1/user/`
    - `GET api/v1/profiles/<str: username>`
    Notice that the request body specifications have changed on the [README](https://github.com/andela/ah-scorpion/blob/ch-Refactor-Code-%23159459856/README.md)

**What are the relevant PivotalTracker stories?**
[159459856](https://www.pivotaltracker.com/story/show/159459856)
